src/DataStructures/Tensor/Expressions/AddSubtract.hpp

```diff
@@ -29,8 +32,142 @@ struct TensorExpression;
 /// \endcond
 
 namespace TensorExpressions {
-
 namespace detail {
+// @{
+/// \ingroup TensorExpressionsGroup
+/// \brief Returns the canonical symmetry of the tensor resulting from
+/// adding or subtracting two tensors, according to their symmetries
+///
+/// \details The canonical symmetry returned follows the convention defined by
+/// ::Symmetry: symmetry values are in ascending order from right to left. If
+/// the convention implemented by ::Symmetry changes, this function will also
+/// need to be updated to match the new convention.
+///
+/// This function treats the two input symmetries as aligned (i.e. each position
+/// of `symm1` and `symm2` corresponds to a shared generic index at that
+/// position). The resultant symmetry is determined as follows: indices that are
+/// symmetric in both input symmetries are also symmetric in the resultant
+/// tensor.
+///
+/// \param symm1 the symmetry of the first tensor being added or subtracted
+/// \param symm2 the symmetry of the second tensor being added or subtracted
+/// \return the canonical symmetry of the tensor resulting from adding or
+/// subtracting two tensors
+template <size_t NumIndices, Requires<(NumIndices >= 2)> = nullptr>
+constexpr std::array<std::int32_t, NumIndices> get_addsub_symm(
+    const std::array<std::int32_t, NumIndices>& symm1,
+    const std::array<std::int32_t, NumIndices>& symm2) {
+  std::array<std::int32_t, NumIndices> addsub_symm{};
```

Comment:
This was a really good catch and would have been nasty to hunt down down the line. Will fix

Reply:
I'm totally happy to use `make_array` regardless of the following, but just for knowledge purposes, I wanted to mention that it seems like based on what I'm reading, this array might already be zero-initialized. My understanding of the following documents might be wrong, but this is what I'm seeing:
- [This](https://en.cppreference.com/w/cpp/container/array) sounds like `std::array` + curly braces does aggregate initialization
- [This](https://en.cppreference.com/w/cpp/language/aggregate_initialization) sounds like `std::array<std::int32_t, N> arr{};` means value-initialization will be used because the initializer list is empty
- [This](https://en.cppreference.com/w/cpp/language/value_initialization) sounds like value-initialization for `std::int32_t` is `0`

Reply:
I'm totally happy to use `make_array` regardless of the following, but just for knowledge purposes, I wanted to mention that it seems like based on what I'm reading, this array might already be zero-initialized. My understanding of the following documents might be wrong, but this is what I'm seeing:
- [This](https://en.cppreference.com/w/cpp/container/array) sounds like `std::array` + curly braces does aggregate initialization
- [This](https://en.cppreference.com/w/cpp/language/aggregate_initialization) sounds like `std::array<std::int32_t, N> arr{};` means value-initialization will be used because the initializer list is empty
- [This](https://en.cppreference.com/w/cpp/language/value_initialization) sounds like value-initialization for `std::int32_t` is `0`

Reply:
It looks like the `{}` does guarantee the value is zero (I recall you linking some documentation in this). I'd still suggest using a more obviously incorrect sentinel value, but up to you :)

Reply:
Yah I ended up looking up documentation on cpp reference and thought I saw that the `int`s would be zero-initialized and then realized that I think I misread it. I saw [this](https://en.cppreference.com/w/cpp/language/aggregate_initialization), which seems to say that for `{}`, value initialization was used before C++11 (I didn't see the "before C++ 11" part at first), and then it says that default initialization is used since C++11. So, I then thought: never mind, sounds like `int`s in a `std::array` aren't guaranteed to be 0 if they are default initialized. My understanding of default initialization of `int`s is that they could hold any value. I removed the links because I think I realized my original interpretation was incorrect.

Either way, I'm happy to use a different sentinel value and will do that!